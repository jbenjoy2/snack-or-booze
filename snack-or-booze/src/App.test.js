import React from 'react';
import App from './App';
import { MemoryRouter } from 'react-router-dom';

import { render } from '@testing-library/react';
import { act } from 'react-dom/test-utils';

function sleep(time) {
	return new Promise((resolve) => setTimeout(resolve, time));
}

it('renders without crashing', () => {
	render(
		<MemoryRouter initialEntries={[ '/' ]}>
			<App />
		</MemoryRouter>
	);
});

it('matches snapshot', () => {
	const { asFragment } = render(
		<MemoryRouter initialEntries={[ '/' ]}>
			<App />
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});

it('can render the /home route', async () => {
	const { getByText } = render(
		<MemoryRouter initialEntries={[ '/' ]}>
			<App />
		</MemoryRouter>
	);

	await act(() => sleep(2000));
	const text = getByText("Welcome to Silicon Valley's premier dive cafe!");
	expect(text).toBeInTheDocument();
});
