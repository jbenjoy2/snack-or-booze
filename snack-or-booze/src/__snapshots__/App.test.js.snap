// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <p
    style="color: white;"
  >
    Loading â€¦
  </p>
</DocumentFragment>
`;
